vcd/resource_vcd_ip_space.go

```diff
@@ -0,0 +1,427 @@
+	}
+
+	ipSpace, err := vcdClient.GetIpSpaceById(d.Id())
+	if err != nil {
+		return diag.Errorf("error finding IP Space by ID '%s': %s", d.Id(), err)
+	}
+
+	ipSpaceConfig.ID = d.Id()
+	_, err = ipSpace.Update(ipSpaceConfig)
+	if err != nil {
+		return diag.Errorf("error updating IP Space: %s", err)
+	}
+
+	return resourceVcdIpSpaceRead(ctx, d, meta)
+}
+
+func resourceVcdIpSpaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space read initiated")
+
+	ipSpace, err := vcdClient.GetIpSpaceById(d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error finding IP Space by ID '%s': %s", d.Id(), err)
+	}
+
+	err = setIpSpaceData(d, ipSpace.IpSpace)
+	if err != nil {
+		return diag.Errorf("error storing IP Space state: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdIpSpaceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space deletion initiated")
+
+	ipSpace, err := vcdClient.GetIpSpaceById(d.Id())
+	if err != nil {
+		return diag.Errorf("error finding IP Space by ID '%s': %s", d.Id(), err)
+	}
+
+	err = ipSpace.Delete()
+	if err != nil {
+		return diag.Errorf("error deleting IP space by ID '%s': %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+// resourceVcdIpSpaceImport has two cases:
+// * Import global (provider level) IP Space - just a name is required
+// * Import Private IP space for an Organization - org-name.ip-space-name is required
+func resourceVcdIpSpaceImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] IP Space import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 1 && len(resourceURI) != 2 {
+		return nil, fmt.Errorf("resource name must be specified as ip-space-name or org-name.ip-space-name")
+	}
+	vcdClient := meta.(*VCDClient)
+
+	var ipSpace *govcd.IpSpace
+
+	if len(resourceURI) == 2 {
+		ipSpaceName := resourceURI[1]
+		orgName := resourceURI[0]
+
+		org, err := vcdClient.GetOrgByName(orgName)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving Org '%s': %s", orgName, err)
+		}
+
+		ipSpace, err = vcdClient.GetIpSpaceByNameAndOrgId(ipSpaceName, org.Org.ID)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving IP Space '%s' in Org '%s': %s", ipSpaceName, orgName, err)
+		}
+		dSet(d, "org_id", org.Org.ID)
+
+	} else {
+		var err error
+		ipSpaceName := resourceURI[0]
+		ipSpace, err = vcdClient.GetIpSpaceByName(ipSpaceName)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving IP Space '%s': %s", ipSpaceName, err)
+		}
+	}
+
+	d.SetId(ipSpace.IpSpace.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func getIpSpaceType(d *schema.ResourceData) (*types.IpSpace, error) {
```

Comment:
Maybe this could have more descriptive name like `getIpSpaceConfig`? I understand the idea, it's just not clear when IP Spaces have their types apart from that

Reply:
Not very consistent through the code, but this is sort of a consistent name we had a soft agreement where. As it is not too consistent I am open for change as well  if you think `getIpSpaceConfig` is better

You would be able to find quite a few functions with names `getXType` (to build a particular type from Terraform schema) and the reverse `setXData` (which uses the type from API to store it in schema.